src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonValueConverterNullable.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Text.Json.Serialization
+{
+    internal class JsonValueConverterNullable<T> : JsonConverter<T?> where T : struct
+    {
+        // It is possible to cache the underlying converter since this is an internal converter and
+        // an instance is created only once for each JsonSerializerOptions instance.
+        private JsonConverter<T> _converter;
+
+        public JsonValueConverterNullable(JsonConverter<T> converter)
```

Comment:
I am not sure here, but should this be `JsonConverter<T?> converter`. I am assuming no, but am having trouble reasoning about what the expectation should be. This is called by the `JsonValueConverterNullableFactory`.

Reply:
`_converter` is not and should not be nullablel. The purpose of the converter is to forward to the non-nullable `_converter`.

If it could be nullable that would imply a `Nullable<Nullable<T>>` which is not supported.